### ISSUE TITLE
A11Y: Make message history accessible in VS Code

### ISSUE THREAD

**Issue by paul-geoghegan** [continuedev/continue#12331]
### Before submitting your bug report

- [ ] I've tried using the "Ask AI" feature on the [Continue docs site](https://docs.continue.dev/) to see if the docs have an answer
- [x] I'm not able to find a related conversation on [GitHub discussions](https://github.com/continuedev/continue/discussions) that reports the same bug
- [x] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug
- [x] I've seen the [troubleshooting guide](https://docs.continue.dev/troubleshooting) on the Continue Docs

### Relevant environment info

```Markdown
- OS: Ubuntu
- Continue version: Any
- IDE version:
- Model:
- config:
  

  
  OR link to agent in Continue hub:
```

### Description

The message history is currently not accessible for screen readers. I will provide a list of the needed fixes:
- Currently it doesn't seem to be in the tab order. This can probably be done by adding a tabindex to each individual message. If the messages are rendered as HTML then only the parent container may need a tabindex as you should be able to navigate a html view with ease if each message has a corisponding heading etc,
- The surounding container needs an aria label like "message history" or similar,
- Any code blocks with buttons for actions need to probably have an aria label or a different roll to distinguish them and their corisponding buttons need to be labeled and should be accessible via tab,
- If possible hook in to the VS Code native chat interface so that settings that allow for progress updates etc to be convayed to screen readers are respected.
It's possible that this is not an exaustive list but it would be a great start.

### To reproduce

1. turn on a screen reader e.g. windows: control+windows+enter, linux: super+alt+s.
2. Open VS Code,
3. Open the continue chat with control+i
4. Notice the inaccessible chat history.

### Log output

```Shell

```

**Comment by drewpoling2** [comment]
I'd like to take a stab at this. I'll start with the tab order and aria labeling for the message container and code block buttons. Will submit a draft PR once I have something to show.
